contribution/translation/prevalidator.php

```diff
@@ -67,282 +69,163 @@ public function get_helper()
+		$finder->files()->in($this->get_helper()->get_root_path() . self::LANGUAGE_PACKAGES);
+
+		if ($finder->hasResults())
 		{
-			// Get all the files present in the uploaded package for the currently iterated type
-			$uploaded_files = $this->lang_filelist($package->get_temp_path());
-			$reference_files = $this->lang_filelist($reference_filepath);
-			ksort($uploaded_files);
-			ksort($reference_files);
+			$latest = ['name' => '', 'x' => 0, 'y' => 0, 'z' => 0];
 
-			switch ($type)
+			foreach ($finder as $file)
 			{
-				case 'language':
-					// The uploaded files array has keys prefixed with the upload path of the contribution
-					// Have it stored in the variable so we can work with it
-					$uploaded_files_prefix = explode('/', key($uploaded_files));
-					$iso_code = $uploaded_files_prefix[2];
-					$uploaded_files_prefix = $package->get_temp_path() . '/' . $uploaded_files_prefix[0];
-
-					// This goes directly to the root of the uploaded language pack, like /upload_path/language/cs/
-					$uploaded_lang_root = $uploaded_files_prefix . '/language/' . $iso_code . '/';
-
-					// Just perform a basic check if the common file is there
-					if (!is_file($uploaded_lang_root . 'common.php'))
-					{
-						return array($this->user->lang('NO_TRANSLATION'));
-					}
+				// Extract the version number
+				preg_match_all('/' . self::BRITISH_ENGLISH . '(\d+)_(\d+)_(\d+)\.zip/', $file->getFilename(), $result);
 
-					// Loop through the reference files
-					foreach ($reference_files as $dir => $files)
-					{
-						// Do not loop through files which are in non-language directories
-						if (strpos($dir, $path) === 0)
-						{
-							// Loop through the files in the language/, language/adm etc. directories
-							foreach ($files as $file)
-							{
-								$exists = true;
-								$uploaded_file_path = str_replace('/en/', '/' . $iso_code . '/', $uploaded_files_prefix . '/' . $dir . $file);
-
-								$ext = strtolower(substr($file, -3));
-								// Require php and txt files
-								if ($ext == 'php' || $ext == 'txt')
-								{
-									if (!is_file($uploaded_file_path))
-									{
-										$error[] = $this->user->lang('MISSING_FILE', str_replace('/en/', '/' . $iso_code . '/', $dir. $file)); // report a missing file
-										$exists = false;
-									}
-								}
-
-								// If the file is a php file and actually exists, no point in checking keys in a nonexistent one
-								if ($ext == 'php' && $exists)
-								{
-									$missing_keys[$dir . $file] = $this->check_missing_keys($reference_filepath . '' .	$dir . $file, $uploaded_file_path);
-								}
-
-								// In the last step we have removed the license and index files if there were any. We'll just put a new one instead
-								$this->add_license_files($uploaded_lang_root . '/LICENSE', $reference_filepath);
-								$this->add_htm_files($uploaded_lang_root, $reference_filepath);
-							}
-						}
-					}
-
-					if (sizeof($missing_keys))
+				if ($result[0])
```

Comment:
As for this whole thing. I was thinking a lot simpler... Like:
```php
$finder->files()
    ->in($this->get_helper()->get_root_path() . self::LANGUAGE_PACKAGES)
    ->name(self::BRITISH_ENGLISH . *')
    ->sortByName();
```
Then just grab the last item in the array, which should be the latest

Reply:
Hmm I'm worried we'll get unexpected results if we don't do the comparison manually. Think of things like:

_3_2_5
_3_2_10

Alphabetically I think _3_2_10 would come first because of the 1, even though it's the higher version.

Reply:
Have you tested that?

Reply:
You're probably right

Reply:
Yeah:

![ch17](https://user-images.githubusercontent.com/2110222/56466306-6fee0600-6442-11e9-9e52-07e841109be1.png)